### ISSUE TITLE
False positive `unfulfilled_lint_expectations` for `enum_glob_use` with `--all-targets` or `--tests`

### ISSUE THREAD

**Issue by lnicola** [rust-lang/rust-clippy#13358]
### Summary

`unfulfilled_lint_expectations` seems to trigger for `enum_glob_use` when either `--all-targets` or `--tests` is used. Yet other lints, like `unreadable_literal` don't behave that way.

https://github.com/rust-lang/rust/issues/130021
https://github.com/rust-lang/rust-analyzer/issues/17685

### Lint Name

enum_glob_use

### Reproducer

I tried this code:

```rust
#![warn(clippy::pedantic)]

#[expect(clippy::enum_glob_use)]
use std::cmp::Ordering::*;

fn main() {
    #[expect(clippy::unreadable_literal)]
    let _ = 1000000000;
    print!("{Equal:?}");
}
```

I saw this happen:

```
$ cargo clippy --all-targets
warning: this lint expectation is unfulfilled
 --> src/main.rs:3:10
  |
3 | #[expect(clippy::enum_glob_use)]
  |          ^^^^^^^^^^^^^^^^^^^^^
  |
  = note: `#[warn(unfulfilled_lint_expectations)]` on by default

warning: `hello` (bin "hello" test) generated 1 warning
    Finished `dev` profile [unoptimized + debuginfo] target(s) in 0.00s
```

I expected to see this happen:

```
$ cargo clippy
    Finished `dev` profile [unoptimized + debuginfo] target(s) in 0.00s
```

### Version

```text
rustc 1.81.0-beta.6 (b5fd9f6f1 2024-08-21)
binary: rustc
commit-hash: b5fd9f6f1061b79c045cc08fe03e00caad536800
commit-date: 2024-08-21
host: x86_64-unknown-linux-gnu
release: 1.81.0-beta.6
LLVM version: 18.1.7
```

**Comment by Urgau** [comment]
The lint emission is gated behind the compilation unit NOT behind a test harness generation, it shouldn't be.
https://github.com/rust-lang/rust-clippy/blob/fb9913ef3be6af354de0e0725d7f82a075d94b89/clippy_lints/src/wildcard_imports.rs#L119-L121

**Comment by junbl** [comment]
Same issue with `missing_docs_in_private_items`, unfulfilled lint on `all_targets` and `tests` and nothing otherwise:

```
> cargo clippy 
    Finished `dev` profile [unoptimized + debuginfo] target(s) in 0.03s
> cargo clippy --all-targets
warning: this lint expectation is unfulfilled
 --> src/main.rs:5:10
  |
5 | #[expect(clippy::missing_docs_in_private_items)]
  |          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  |
  = note: `#[warn(unfulfilled_lint_expectations)]` on by default

warning: `playground` (bin "playground" test) generated 1 warning
    Finished `dev` profile [unoptimized + debuginfo] target(s) in 0.00s

```

with this code:
```rust
#![deny(clippy::missing_docs_in_private_items)]

#[expect(clippy::missing_docs_in_private_items)]
const WHO_AM_I: () = ();
```

**Comment by dragonrider7225** [comment]
> Same issue with both missing_docs and clippy::missing_docs_in_private_items, unfulfilled lint on all_targets and tests and nothing otherwise:

It's worth noting that in the case of `missing_docs`, this issue has been resolved as of rustc 1.83.0: [https://github.com/rust-lang/rust/pull/130025].

**Comment by sww1235** [comment]
seeing the same issue as @aksh1618 on the below version when running with `cargo clippy --all-targets`
```
$ rustc --version && cargo --version && cargo clippy --version
rustc 1.91.1 (ed61e7d7e 2025-11-07)
cargo 1.91.1 (ea2d97820 2025-10-10)
clippy 0.1.91 (ed61e7d7e2 2025-11-07)
```
